### ISSUE TITLE
paypal reference transaction / recurring payment

### ISSUE THREAD

**Issue by pawelwaniek** [PAYONE-GmbH/magento-1#77]
For implementing a recurring payment for paypal I was told by PAYONE to set the payment parameters "recurring" and "customer_is_present".
Unfortunately this is not possible, because the properties doesn't exists in the Library (Payone_Api_Request_Parameter_Authorization_PaymentMethod_Wallet)
I know this feature is not supported yet, but without the properties it's impossible to implement it even by our own.

A generic method like "setValue()" exists, but there is a check if the property exists lib/Payone/Api/Request/Parameter/Abstract.php:104

It would be nice if this properties could be implemented. To be honest I'm not familiar enough with the Lib to implement it by my own. Otherwise I would create a PR

**Comment by fjbender** [comment]
I guess it would just be the easiest (and fastest) way just to add the needed properties to `Payone_Api_Request_Parameter_Authorization_PaymentMethod_Wallet` and see if it works :smile: 

```php
class Payone_Api_Request_Parameter_Authorization_PaymentMethod_Wallet
    extends Payone_Api_Request_Parameter_Authorization_PaymentMethod_Abstract
{
    /**
     * @var string
     */
    protected $wallettype = NULL;
    /**
     * @var string
     */
    protected $successurl = NULL;
    /**
     * @var string
     */
    protected $errorurl = NULL;
    /**
     * @var string
     */
    protected $backurl = NULL;

    protected $paydata = NULL;

    /**
     * @var string
     */
    protected $recurring = NULL;

    /**
     * @var string
     */
    protected $customer_is_present = NULL;

    // TODO: add appropriate setters and getters
```

If that doesn't work I'd like to take a look again.

**Comment by pawelwaniek** [comment]
After clarification with PAYONE the parameter is called "recurrence" not "recurring"

PR is created: https://github.com/PAYONE-GmbH/magento-1/pull/80

**Comment by Schrank** [comment]
PR #80 is merged, this ticket can be closed.

**Comment by fjbender** [comment]
Fixed with #80
